presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -193,25 +194,6 @@ public void put(Expression expression, Expression rewritten)
         expressionToExpressions.put(translateNamesToSymbols(expression), rewritten);
     }
 
-    public void addIntermediateMapping(Expression expression, Expression rewritten)
```

Comment:
do you remember what was the semantic of this?

Reply:
It was really a hack that should’ve never been added. It was there to support query parameters, but there was a better way to do it.

The reason I have to remove it is that with the new IR there’s no rewriting of expressions to the same “language”. It has to be a straight translation between AST language to IR language.

Reply:
> It was really a hack that should’ve never been added. It was there to support query parameters, but there was a better way to do it.

https://github.com/prestodb/presto/issues/7989 is a proposal to cleanup parameters processing. Essentially for queries with parameters we could cache IR plan where `Parameter` expressions are later replaced with proper constant (e.g: by some desugarer or rule). This would allow to remove dependencies on `ParameterRewriter` too.